artifacts/fedora/fedora.go

```diff
@@ -112,21 +116,30 @@ func (f *fedoraGatherer) Gather() ([][]api.Artifact, error) {
 		return nil, fmt.Errorf("error getting releases: %v", err)
 	}
 
-	artifacts := [][]api.Artifact{}
-	for i, release := range releases {
-		if f.releaseMatches(&releases[i]) {
-			artifacts = append(artifacts,
-				[]api.Artifact{
-					New(
-						release.Version,
-						map[string]string{
-							common.DefaultInstancetypeEnv: "u1.small",
-							common.DefaultPreferenceEnv:   "fedora",
-						},
-					),
-				},
-			)
+	versions := map[string][]Release{}
```

Comment:
It's not clear to me why any of this needed to change, it was already capturing all of the required artifacts right?

If the change is required please document why in the commit message.

Reply:
Yes I'll update commit message to document it, I was struggling myself to modify it. Thanks to @0xFelix I could understand how it works.

I replied below in this PR why it's needed:
> The changes to `Gather()` required because we'd like to gather also arm64 images. Before that we had:
> 
> ```
> Entry: [0x1400064a2d0 0x1400064a320]
> Artifact: &{39 x86_64 Cloud 0x10763f6e0 map[INSTANCETYPE_KUBEVIRT_IO_DEFAULT_INSTANCETYPE:u1.small INSTANCETYPE_KUBEVIRT_IO_DEFAULT_PREFERENCE:fedora]}
> Entry: [0x1400064a2d0 0x1400064a320]
> Artifact: &{38 x86_64 Cloud 0x10763f6e0 map[INSTANCETYPE_KUBEVIRT_IO_DEFAULT_INSTANCETYPE:u1.small INSTANCETYPE_KUBEVIRT_IO_DEFAULT_PREFERENCE:fedora]}
> ```
> 
> And now we'll have:
> 
> ```
> Entry: [0x1400064a2d0 0x1400064a320]
> Artifact: &{39 x86_64 Cloud 0x10763f6e0 map[INSTANCETYPE_KUBEVIRT_IO_DEFAULT_INSTANCETYPE:u1.small INSTANCETYPE_KUBEVIRT_IO_DEFAULT_PREFERENCE:fedora]}
> Artifact: &{39 aarch64 Cloud 0x10763f6e0 map[INSTANCETYPE_KUBEVIRT_IO_DEFAULT_INSTANCETYPE:u1.small INSTANCETYPE_KUBEVIRT_IO_DEFAULT_PREFERENCE:fedora]}
> Entry: [0x1400064a2d0 0x1400064a320]
> Artifact: &{38 x86_64 Cloud 0x10763f6e0 map[INSTANCETYPE_KUBEVIRT_IO_DEFAULT_INSTANCETYPE:u1.small INSTANCETYPE_KUBEVIRT_IO_DEFAULT_PREFERENCE:fedora]}
> Artifact: &{38 aarch64 Cloud 0x10763f6e0 map[INSTANCETYPE_KUBEVIRT_IO_DEFAULT_INSTANCETYPE:u1.small INSTANCETYPE_KUBEVIRT_IO_DEFAULT_PREFERENCE:fedora]}
> ```

Reply:
Right apologies I see it now, we are changing the returned slice to contained grouped Artifacts for each release, so something from this to this:

```
[][]Artifact{
    {
        Artifact{39, x86_64},
    },
    {
        Artifact{39, aarch64},
    },
    {
        Artifact{40, x86_64},
    },
    {
        Artifact{40, aarch64},
    },
}

[][]Artifact{
    {
        Artifact{39, x86_64},
        Artifact{39, aarch64},
    },
    {
        Artifact{40, x86_64},
        Artifact{40, aarch64},
    },
}
```

I don't see the need to break this into two loops but I'll not block on that choice. I think this warrents a code comment FWIW.